EthersJs/accelerating-development-with-ethersjs.md

```diff
@@ -0,0 +1,242 @@
+
+```
+<form [formGroup]="transactionForm" (ngSubmit)="sendTransaction(transactionForm.value)">
+        <mat-form-field appearance="outline">
+        <mat-label>To</mat-label>
+        <input formControlName="toAddress" matInput placeholder="0xAddress">
+      </mat-form-field>
+      <mat-form-field appearance="outline">
+        <mat-label>Amount</mat-label>
+        <input formControlName="etherAmount" matInput placeholder="1">
+        <span matSuffix>Ether</span>
+      </mat-form-field>
+        <button mat-raised-button color="secondary">Send Transaction</button>
+    </form>
+```
+
+In your class, import the formModules and create the base form information.
+
+```
+import { Component, OnInit } from '@angular/core';
+import { ethers } from 'ethers';
+import { FormBuilder, FormGroup, NgForm } from '@angular/forms'; 
+
+@Component({
+  selector: 'app-wallet',
+  templateUrl: './wallet.component.html',
+  styleUrls: ['./wallet.component.scss']
+})
+export class WalletComponent implements OnInit {
+
+    publickey: string;
+    privatekey: string;
+    transactionForm: FormGroup;
+    toAddress: string;
+    etherAmount: string;
+    wallet: any;
+    provider: any;
+
+  constructor(private fb: FormBuilder) { 
+      this.transactionForm = fb.group({
+          'toAddress': [null],
+          'etherAmount': [null]
+      });
+  }
+
+  ngOnInit() {
+      this.provider = ethers.getDefaultProvider('homestead');;
+  }
+
+  onSubmit() {
+      const randomWallet = ethers.Wallet.createRandom();
+      this.publickey = randomWallet.address;
+      this.privatekey = randomWallet.privateKey;
+      this.wallet = new ethers.Wallet(this.privatekey, this.provider)
+  }
+
+}
+```
+
+In order to send a transaction, Ethersjs provides a simple sendTransaction method for all wallets. Add a sendTransaction method to the wallet class, inputting the form.
+
+```
+sendTransaction(form: NgForm) {
+    let transaction = {
+      to: form.toAddress,
+      value: ethers.utils.parseEther(form.etherAmount)
+    }
+
+    this.wallet.sendTransaction(transaction)
+        .then((tx) => {
+            console.log(tx);
+        })
+}
+```
+
+We first create the transaction object, giving where the transaction is going in the `to` field. `value` denotes how much ether, default in the units Wei, is being sent to the address mentioned. We use the parseEther util provided by ethers to easily convert from Ether to Wei. After creating the transaction object, we use our wallet we created before to send the transaction. In this implementation, the console logs the transaction receipt.
+
+Signing a message in a transaction is very useful for a number of reasons. A signed message can be used to verify that the owner is the address provided, and this allows smart contracts and other entities to verify the truth in a trustless system. To sign a message using Ethers, you use the signMessage function on your message.
+
+```
+wallet.signMessage(message);
```

Comment:
Do readers need to add this somewhere? If so, where?

Reply:
Removed in an upcoming commit. Decided signing was too much of an aside to include. You can let me know if you think otherwise.